### DISCUSSION TITLE
Make nfpm bindir optional or support multiple bindir (one per built binary)

### DISCUSSION THREAD

**Discussion by jsirianni** [goreleaser/goreleaser#2879] [Q&A] [answered]
Goreleaser has a wrapper parameter [bindir](https://github.com/goreleaser/goreleaser/blob/65289482176b368365560cfb814f07876fde2a3a/internal/pipe/nfpm/nfpm.go#L48) that is used to instruct the nfpm package where to place the binaries previously built by Goreleaser's `build` stage. This is great if you have a single binary, or multiple binaries that should live in the same directory.

This is not ideal if you have multiple binaries that must live in different directories.

### Describe the solution you'd like

I have a fork that skips the line linked above if bindir is set to `SKIP`, this is okay but not great. I use the `contents` config to copy my multiple binaries to their independent locations. Ideally, `bindir` would be defined per binary, or we would have the ability to skip it if already being handled by `contents`.

### Describe alternatives you've considered

I've considered using the post scripts to move the binaries around, but I would prefer to not do this. 

Right now, I am just using bindir as intended, which pollutes my system with extra binaries that will never be called upon (because they exists elsewhere from the `contents` config.

I could use symlinks, however, I would prefer to not stray from how the project expects it's structure.

### Additional context

I am not working on a greenfield project, so I do not have the luxury of changing where the binaries are located. I have a binary in directory A (IE: `/opt/agent/binary`) and a second binary in directory B (IE: `/opt/agent/sub_dir/current/binary`). 

Thank you for considering this, Goreleaser has been a great tool so far. 

---

_EDIT by @caarlos0 to remove the checkboxes et al after migrating from issue to discussion._

**Comment by caarlos0** [comment, marked as answer]
you can set `nfpms.meta=true` and manually add the binaries

```yaml
nfpms:
- meta: true
  contents:
  - src: './dist/foo_{{ .Os }}_{{ .Arch }}' # check your ./dist to see the paths et al
    dst: /bin/foo
    file_info:
      mode: 0755
```

does that work?

**Comment by jsirianni** [comment]
@caarlos0 that worked perfectly, can't believe I missed that option. Thanks!

**Reply by caarlos0** [reply]
amazing, no problem @jsirianni , glad I could help!  ✨
